### PR TITLE
Remove JobQueueManager

### LINKED ISSUE #6285: Remove JobQueueManager
https://github.com/equinor/ert/blob/3da103e6e410d8d166c024f1fb661bd09d64217b/src/ert/job_queue/job_queue_manager.py#L17-L18

### PR DESCRIPTION
**Pull request by berland**
It was only a thin wrapper with no value left. Move its functionality down to the JobQueue object.

**Issue**
Resolves #6285 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/job_queue/queue.py [hunk 1]
```diff
@@ -49,6 +49,11 @@
 logger = logging.getLogger(__name__)
 
 LONG_RUNNING_FACTOR = 1.25
+"""If STOP_LONG_RUNNING is true, realizations taking more time than the average
```
Comment (jonathan-eq) [inline src/ert/job_queue/queue.py:52]: Great addition!

Comment (berland) [reply src/ert/job_queue/queue.py:52]: Just realized that these attribute docstrings are not standardized, and cannot be picked up by any tool.

File: src/ert/simulator/simulation_context.py [hunk 2]
```diff
@@ -155,28 +152,28 @@ def __len__(self) -> int:
 
     def isRunning(self) -> bool:
         # TODO: Should separate between running jobs and having loaded all data
-        return self._sim_thread.is_alive() or self._queue_manager.isRunning()
+        return self._sim_thread.is_alive() or self._job_queue.is_active()
 
     def getNumPending(self) -> int:
-        return self._queue_manager.getNumPending()
+        return self._job_queue.count_status(JobStatus.PENDING)  # type: ignore
```
Comment (jonathan-eq) [inline src/ert/simulator/simulation_context.py:158]: What is the reason for all the `type: ignore`? Will this be fixed in a separate PR/issue?

**Review by jonathan-eq — CHANGES_REQUESTED** [review]

File: src/ert/simulator/simulation_context.py (hunk 2 above)
Comment (berland) [reply src/ert/simulator/simulation_context.py:158]: These are due to difficulties with integrating the typing with the C-code and the custom `cwrap` (not Pybind). It will go away as soon as we have thrown away the C-code..

File: src/ert/job_queue/queue.py [hunk 3]
```diff
@@ -211,11 +217,9 @@ def launch_jobs(self, pool_sema: Semaphore) -> None:
                 max_submit=self.max_submit,
             )
 
-    def execute_queue(
-        self, pool_sema: Semaphore, evaluators: Optional[Iterable[Callable[[], None]]]
-    ) -> None:
+    def execute_queue(self, evaluators: Optional[Iterable[Callable[[], None]]]) -> None:
         while self.is_active() and not self.stopped:
-            self.launch_jobs(pool_sema)
+            self.launch_jobs(self._pool_sema)
```
[the remark this answers is not in the document]
Comment (berland) [reply src/ert/job_queue/queue.py:222]: Yes it can. Well spotted.

File: src/ert/simulator/simulation_context.py (hunk 2 above)
Comment (jonathan-eq) [reply src/ert/simulator/simulation_context.py:158]: Okay, great!

Comment (berland) [reply src/ert/simulator/simulation_context.py:158]: Nope, we need it to retain the possibility to set it when running from _builder/_legacy.py

File: src/ert/job_queue/queue.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (berland) [reply src/ert/job_queue/queue.py:222]: Nope, we need it to retain the possibility to set it when running from _builder/_legacy.py

**Review by jonathan-eq — APPROVED** [review]
Very good, very nice!
